Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A `Stream` implementation that buffers data in memory.
+/// The buffer size depends on the amount of data written and read.
+/// Note: This class is thread-safe.
+public class BufferedStream: Stream {
+
+    /// The length of the stream, if known.
+    public var length: Int?
+
+    /// The current position in the stream.
+    public var position: Data.Index
+
+    /// Whether the stream is empty.
+    public var isEmpty: Bool {
+        return buffer.isEmpty
+    }
+
+    /// Whether the stream is seekable.
```

Comment:
Imo we should prefer a format of `Returns a if x,y,z otherwise returns b` for doc comments on simple booleans. 
Such as `Returns true if the stream is seekable, otherwise returns false. `
Imo this is a bit more clear.